doc/mgmt/Management Framework.md

```diff
@@ -651,63 +845,111 @@ REST server will accept concurrent requests. Translib provides appropriate locki
+GET    | /models/yang/{filename}                        | YANG download
 
-###### 3.2.2.4.9 RESTCONF Query Parameters
+###### 3.2.2.4.8.1 RESTCONF root
 
-RESTCONF Query Parameters will be supported in future release. All query parameters will be ignored by REST server in this release.
+Server supports RESTCONF root path discovery through "GET /.well-known/host-meta" API
+as described in [RFC8040, section 3.1](https://tools.ietf.org/html/rfc8040#page-18).
+RESTCONF root path is "/restconf".
 
-###### 3.2.2.4.10 RESTCONF Operations
+###### 3.2.2.4.8.2 Yang library version
 
-RESTCONF operations via YANG RPC are not supported in this release. They can be supported in future releases.
+REST server supports "GET /restconf/yang-library-version" API to advertise YANG library version.
+Response will indicate version "2016-06-21" as described in [RFC8040, section 3.3.3](https://tools.ietf.org/html/rfc8040#section-3.3.3).
+This advertises that the server supports [RFC7895](https://tools.ietf.org/html/rfc7895) compliant
+YANG library operations.
 
-###### 3.2.2.4.11 RESTCONF Notifications
+###### 3.2.2.4.8.3 YANG module library
 
-RESTCONF Notification are not supported by framework. Clients can use gNMI for monitoring and notifications.
+REST server allows clients to discover and download all YANG modules supported by the server
+via "GET /restconf/data/ietf-yang-library:modules-state" API. Response data includes YANG module
+information as per [RFC7895](https://tools.ietf.org/html/rfc7895) requirements.
 
-###### 3.2.2.4.12 Authentication
+REST server allows clients to download the YANG files via "GET /models/yang/{filename}" API.
+YANG module library response includes full download URL for every YANG module entry.
 
-REST server will support below 3 authentication modes.
+Note - RFC7895 has been recently obsoleted by [RFC8525](https://tools.ietf.org/html/rfc8525).
+It supports YANG library with multi data stores and data store schemas. However these features
+are not available in SONiC. Hence REST server does not implement RFC8525 YANG library APIs (for now).
 
-* No authentication
-* TLS Certificate authentication
-* Username/password authentication
+###### 3.2.2.4.8.4 RESTCONF capabilities
+
+REST server supports "GET /restconf/data/ietf-restconf-monitoring:restconf-state/capabilities" API to
+advertise its capabilities as described in [RFC8040, section 9.1](https://tools.ietf.org/html/rfc8040#section-9.1).
+REsponse includes below mentioned capability information.
+
+    urn:ietf:params:restconf:capability:defaults:1.0?basic-mode=report-all
+
+###### 3.2.2.4.9 RESTCONF Query Parameters
 
-Only one mode can be active at a time. Administrator can choose the authentication mode through ConfigDB REST_SERVER table entry. See [DB Schema](#3_2_2_4_14-db-schema) section.
+RESTCONF Query Parameters will be supported in future release. All query parameters will be ignored by REST server in this release.
```

Comment:
Does the design support pagination ? The possible usecase could be - for the long API responses, basically to consumek by any UX applications

Reply:
RESTCONF does not have provision for pagination. So the Mgmt REST server does not support any "page cookie" like query parameters. We plan to add HTTP/1.1 chunked transfer or HTTP/2 stream support in future releases. This should ease the server load while servicing large response payload.

App developer can provide old style cookie based pagination through yang custom RPCs. Clients can specify page cookie in request body and not query parameter. However this will be an app specific API and not generic server-wide API.